src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -427,8 +432,22 @@ private void onSessionInitiated(SessionMessage sessionMessage) {
   }
 
   private void logIn(String username, String password) {
-    String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
-    writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    try {
+      String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
+      writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    } catch (IOException e) {
+      onUIDNotFound(e);
```

Comment:
`IOException` doesn't mean "not found". It could also be access denied, non-readable, failure when reading etc. So the method should be renamed. Thus, the method should be renamed.